ci-operator/templates/openshift/installer/cluster-launch-installer-metal-e2e.yaml

```diff
@@ -299,20 +299,35 @@ objects:
         echo "Removing bootstrap host from control plane api pool"
         (cd /tmp/artifacts/terraform && terraform apply -auto-approve=true -var=bootstrap_dns="false")
 
-        echo "Approving any pending CSR requests"
-        oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs oc adm certificate approve
+        function approve_csrs() {
+          while true; do
+            if [[ ! -f /tmp/install-complete ]]; then
+              oc get csr -ojson | jq -r '.items[] | select(.status == {} ) | .metadata.name' | xargs --no-run-if-empty oc adm certificate approve
+              sleep 15 & wait
+              continue
+            else
+              break
+            fi
+          done
+        }
 
-        until oc get configs.imageregistry.operator.openshift.io cluster >/dev/null
-        do
-          echo "waiting for configs.imageregistry.operator.openshift.io cluster to exit failed. Retrying in 1 minute..."
-          sleep 1m
-        done
+        function update_image_registry() {
+          while true; do
+            sleep 10;
+            oc get configs.imageregistry.operator.openshift.io/cluster > /dev/null && break
+          done
+          oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"storage":{"emptyDir":{}}}}'
+        }
+
+        echo "Approving pending CSRs"
+        approve_csrs &
 
-        oc patch configs.imageregistry.operator.openshift.io cluster --type merge --patch '{"spec":{"storage":{"emptyDir":{}}}}'
+        update_image_registry &
```

Comment:
If the pattern is to echo some progress message before taking action, should we have `echo 'Updating image registry storage...'` or something similar before this line?  I'm also fine just having the script silently do its job, but it feels odd to have this `update_image_registry` appear to happen under an `Approving pending CSRs` header.